Add documentation on how NGINX Agent works and Install instructions (#74)
nginxinc/nginx-amplify-doc: content/how-agent-works/agent-overview.md

NGINX Amplify Agent is a compact application written in Python. Its role is to collect various metrics and metadata and send them securely to the backend for storage and visualization.

> You will need to install the Amplify Agent on all hosts that you wish to monitor.

Reviewer: ```suggestion
You need to install the Amplify Agent on all hosts you want to monitor.
```

> After installation, the agent will automatically start to report metrics. You should see the real-time metrics data in the NGINX Amplify web interface in about 60 seconds.

Reviewer: This is why we should use full product names. In this pull request, the term "agent" refers to both Amplify Agent and NGINX Agent. If it's confusing me, it'll confuse customers.

> After installation, the agent will automatically start to report metrics. You should see the real-time metrics data in the NGINX Amplify web interface in about 60 seconds.

Reviewer: ```suggestion
Once you install the Amplify Agent, it will send metrics immediately. You can expect to see real-time metrics in the NGINX Amplify web interface within about a minute.
```

  3. [PHP-FPM]({{< relref "/metrics-metadata/other-metrics.md#php-fpm-metrics" >}})
  4. [MySQL]({{< relref "/metrics-metadata/other-metrics.md#mysql-metrics" >}})

> The agent considers an NGINX instance to be any running NGINX master process that has a unique path to the binary or a unique configuration.

Reviewer: ```suggestion
The NGINX Amplify Agent identifies an NGINX instance as any running NGINX master process with either a unique binary path or a unique configuration.
```
